core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/V1WorkspaceMigrationMonitorSpec.scala

```diff
@@ -82,4 +84,37 @@ class V1WorkspaceMigrationMonitorSpec
     }
   }
 
+  "claimAndConfigureGoogleProject" should "return a valid database operation" in {
+    val spec = new WorkspaceServiceSpec()
+    spec.withTestDataServices { services =>
+      spec.runAndWait {
+        DBIO.seq(
+          spec.workspaceQuery.createOrUpdate(spec.testData.v1Workspace),
+          V1WorkspaceMigrationMonitor.schedule(spec.testData.v1Workspace)
+        )
+      }
+
+      val (_, _, dbOp) = Await.result(
+        services.slickDataSource.database
+          .run {
+            V1WorkspaceMigrationMonitor.migrations
+              .filter(_.workspaceId === spec.testData.v1Workspace.workspaceIdAsUUID)
+              .result
+          }
+          .map(_.head)
+          .flatMap { attempt =>
+            V1WorkspaceMigrationMonitor.claimAndConfigureNewGoogleProject(
+              attempt,
+              services.workspaceService,
+              spec.testData.v1Workspace,
+              spec.testData.billingProject
+            )
+          },
+        5.seconds // really should be instant if we've mocked correctly?
+      )
+
+      spec.runAndWait(dbOp) shouldBe ()
```

Comment:
Is there a reason why you decided not to write an assertion on the migration record that gets updated as part of `claimAndConfigureNewGoogleProject`?

Reply:
I dind't really want to expose database internals outside of the v1 workspace monitor. I certanly could.